pkg/ipmi/ipmi.go

```diff
@@ -173,3 +192,58 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (i *IPMI) setsysinfo(data *setSystemInfoReq) error {
+	req := &req{}
+	req.msg.cmd = _SET_SYSTEM_INFO_PARAMETERS
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.dataLen = 18 // size of setSystemInfoReq
+	req.msg.data = unsafe.Pointer(data)
+
+	if _, err := i.sendrecv(req); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func strcpyPadded(dst []byte, src string, n int) {
```

Comment:
I think the last parameter is not needed. n is the same as len(dst), isn't it?

Reply:
Yes this function is only served for this purpose, n is always the same as len(dst), will update.